pdp/flags_mapper_pca_test.go

```diff
@@ -128,10 +128,8 @@ func TestFlagsMapperPCAMarshal(t *testing.T) {
 		Algorithm: firstApplicableEffectPCA{},
 	}
 	alg := makeMapperPCA(policies, algParam)
-	b, err := json.Marshal(alg)
-	if err != nil {
-		t.Fatalf("Expected no marshaling error but got: %s", err)
-	} else if expectEmptyFlagMapperPCAJSON != string(b) {
+	b, _ := json.Marshal(alg)
+	if expectEmptyFlagMapperPCAJSON != string(b) {
```

Comment:
Sorry if my original comment was misleading..

There's nothing wrong with the original code:
```
b, err := json.Marshal(alg)
if err  != nil {
```
etc

The main point is:
```
if expectEmptyFlagMapperPCAJSON != string(b) { 
```
should be:
```
if string(b) != expectEmptyFlagMapperPCAJSON  { 
```
instead.